snuba/admin/migrations_policies.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from functools import wraps
+from typing import Any, Callable, Dict
+
+from flask import Response, jsonify, make_response
+from werkzeug.routing import BaseConverter
+
+from snuba import settings
+from snuba.migrations.policies import (
+    AllMigrationsPolicy,
+    MigrationPolicy,
+    NoMigrationsPolicy,
+    NonBlockingMigrationsPolicy,
+)
+from snuba.migrations.runner import MigrationKey
+
+policy_map = {
+    "AllMigrationsPolicy": AllMigrationsPolicy(),
+    "NoMigrationsPolicy": NoMigrationsPolicy(),
+    "NonBlockingMigrationsPolicy": NonBlockingMigrationsPolicy(),
+}
+
+
+class MigrationActionConverter(BaseConverter):
+    regex = r"(?:run|reverse)"
+
+
+def check_migration_perms(f: Callable[..., Response]) -> Callable[..., Response]:
```

Comment:
maybe we can add a docstring here ? maybe just explaining that this is to be used for all migrations endpoints and checks both group access, but then also if a migration is provided, whether or not there is access to run/reverse

Reply:
added